amlb/resources.py

```diff
@@ -81,7 +81,8 @@ def framework_definition(self, name):
         :param name:
         :return: name of the framework as defined in the frameworks definition file
         """
-        framework = self._frameworks[name.lower()]
+        lname = name.lower()
+        framework = next((f for n, f in self._frameworks if n.lower() == lname), None)
```

Comment:
Just thinking out loud here, would
```suggestion
        framework = {n.lower(): f for n, f in self._frameworks}.get(lname)
```
be simpler? I am not convinced.
(I realize in terms of optimization it is worse, but given the context it should be optimized for devs not compute)

Reply:
next is better here, it stops on first found. Building the entire lookup dict is unnecessary.

Reply:
> next is better here, it stops on first found. Building the entire lookup dict is unnecessary.

is what I alluded to with

> (I realize in terms of optimization it is worse, but given the context it should be optimized for devs not compute)

Reply:
> do you find it hard to read?

Not really, I've seen it before but perhaps just not enough, because of the generator syntax the first thought is that we are filtering rather than looking up one element. But that's all there is to it, so either style works for me.